csrc/scheduler/registry.cpp

```diff
@@ -170,7 +170,20 @@ bool checkCanSchedule(
     Fusion* fusion,
     SchedulerRuntimeInfo& runtime_info,
     HeuristicSummary* data_cache = nullptr) {
+
+  // ExprEval scheduler only requires `canScheduleCompileTime` check and should not use this fn.
+  // The following checks build the computeAt map that do not work with SDPAOp.
+  NVF_ERROR(SchedulerType::heuristicType() != ScheduleHeuristic::ExprEval);
+  
   FusionGuard fg(fusion);
+
+  // Fusions with `MatmulOp/LinearOp/SdpaOp` are only accepted in `ExprEval` scheduler, all other schedulers should reject them.
```

Comment:
What about matmul scheduler?

Reply:
You're right, I missed this function has `MmaOp` as well.
Let me correct it, thanks for catching this.

Reply:
I will separate these _ExprEval_ ops from `MmaOp` and check them separately.

Reply:
Yes, the updated changes include that.